velox/type/Filter.cpp

```diff
@@ -490,35 +490,65 @@ std::unique_ptr<Filter> MultiRange::mergeWith(const Filter* other) const {
       return this->clone(/*nullAllowed=*/false);
     case FilterKind::kDoubleRange:
     case FilterKind::kFloatRange:
-    case FilterKind::kBytesRange:
-    case FilterKind::kBytesValues:
-      // TODO Implement
+      // TODO: Implement
       VELOX_UNREACHABLE();
+    case FilterKind::kBytesValues:
+    case FilterKind::kBytesRange:
```

Comment:
I find this logic very hard to follow. Let's split kBytesXxx case from kMultiRange. We can then see if common logic can be factored out into helper methods.

Reply:
Ah ok. The logic is exactly the same.

What I do here is if kMultiRange I extract all the filters in a vector, otherwise the new vector otherFilters has just one element.

Post that its the same logic from line 517-551

Reply:
I wonder if it would be simpler if BytesValues::mergeWith implemented merging with kMultiRange directly. The result would always be a kBytesValues filter (or kAlwaysFalse or kIsNull), right?

Reply:
I think I understand how this works now.

Reply:
Thats right. BytesValues::mergeWith implemented merging with kMultiRange will always return a BytesValues. Should I go ahead and make that change too?

Reply:
Up to you. Whatever you think is best.

Reply:
I will leave it here. I think the code doesnt duplicate if we do this.

Will address the nits, and create the fbsource diff and land this PR.

Thanks!